src/main/java/org/araymond/joal/core/bandwith/weight/PeersAwareWeightCalculator.java

```diff
@@ -4,15 +4,9 @@
 
 public class PeersAwareWeightCalculator {
     public double calculate(final Peers peers) {
-        if (peers.getSeeders() == 0) {
-            return 0.0;
-        }
         final double leechersRatio = peers.getLeechersRatio();
-        if (leechersRatio == 0) {
-            return 0.0;
-        }
-        return leechersRatio * 100
-                * (peers.getSeeders() * leechersRatio)
-                * (((double) peers.getLeechers()) / peers.getSeeders());
+        return (peers.getSeeders() == 0 || leechersRatio == 0)
+                ? 0.0
+                : leechersRatio * 100 * leechersRatio * peers.getLeechers();
```

Comment:
Note `peers.getSeeders()` cancels out, thus the division operation was not needed.

Reply:
🤣 Over-engineering as it's finest !